middleware/service/source/manager/state.cpp

```diff
@@ -636,10 +636,15 @@ namespace casual
          // remove
          local::remove_services( *this, instance, message.services.remove);
 
-         // find all potentially pending.
+         // find all potentially pending that might be enabled by the new service(s)
          auto [ keep, remove] = algorithm::stable::partition( pending.lookups, [&]( auto& pending)
          {
-            if( auto found = algorithm::find( services, pending.request.requested))
+            auto is_requested_concurrent_service = [ &]( const auto& service)
+            {
+               return service.second.is_concurrent() && service.first == pending.request.requested;
+            };
```

Comment:
Hmm, is this totally safe? The service could be _sequential_ also, right? If so, couldn't we get some problems?
I think we need to check  `if( is-concurrent && ! is-sequential && service is the one we want) ` or something like: `service.second.is_only_concurrent()` ?

Reply:
Yes, I think you're right